In [ ]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns

import chart_studio.plotly as py
import plotly.graph_objs as go
import plotly.express as px

In [ ]:
data = pd.read_csv('../../data/AmesHousing.csv', index_col='Order', low_memory=False)
print(data.shape)
data.head()


(2930, 81)


PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street Alley  \
Order                                                                          
1      526301100           20        RL         141.0     31770   Pave   NaN   
2      526350040           20        RH          80.0     11622   Pave   NaN   
3      526351010           20        RL          81.0     14267   Pave   NaN   
4      526353030           20        RL          93.0     11160   Pave   NaN   
5      527105010           60        RL          74.0     13830   Pave   NaN   

      Lot Shape Land Contour Utilities  ... Pool Area Pool QC  Fence  \
Order                                   ...                            
1           IR1          Lvl    AllPub  ...         0     NaN    NaN   
2           Reg          Lvl    AllPub  ...         0     NaN  MnPrv   
3           IR1          Lvl    AllPub  ...         0     NaN    NaN   
4           Reg          Lvl    AllPub  ...         0     NaN    NaN   
5           IR1          Lvl    AllPub  ...         0     NaN  MnPrv   

      Misc Feature Misc Val Mo Sold Yr Sold  Sale Type  Sale Condition  \
Order                                                                    
1              NaN        0       5    2010        WD           Normal   
2              NaN        0       6    2010        WD           Normal   
3             Gar2    12500       6    2010        WD           Normal   
4              NaN        0       4    2010        WD           Normal   
5              NaN        0       3    2010        WD           Normal   

       SalePrice  
Order             
1         215000  
2         105000  
3         172000  
4         244000  
5         189900  

[5 rows x 81 columns]

Total number of Columns: 2930 -- Total Number of Rows: 81

In [20]:
data.columns

Index(['PID', 'MS SubClass', 'MS Zoning', 'Lot Frontage', 'Lot Area', 'Street',
       'Alley', 'Lot Shape', 'Land Contour', 'Utilities', 'Lot Config',
       'Land Slope', 'Neighborhood', 'Condition 1', 'Condition 2', 'Bldg Type',
       'House Style', 'Overall Qual', 'Overall Cond', 'Year Built',
       'Year Remod/Add', 'Roof Style', 'Roof Matl', 'Exterior 1st',
       'Exterior 2nd', 'Mas Vnr Type', 'Mas Vnr Area', 'Exter Qual',
       'Exter Cond', 'Foundation', 'Bsmt Qual', 'Bsmt Cond', 'Bsmt Exposure',
       'BsmtFin Type 1', 'BsmtFin SF 1', 'BsmtFin Type 2', 'BsmtFin SF 2',
       'Bsmt Unf SF', 'Total Bsmt SF', 'Heating', 'Heating QC', 'Central Air',
       'Electrical', '1st Flr SF', '2nd Flr SF', 'Low Qual Fin SF',
       'Gr Liv Area', 'Bsmt Full Bath', 'Bsmt Half Bath', 'Full Bath',
       'Half Bath', 'Bedroom AbvGr', 'Kitchen AbvGr', 'Kitchen Qual',
       'TotRms AbvGrd', 'Functional', 'Fireplaces', 'Fireplace Qu',
       'Garage Type', 'Garage Yr Blt', 'Garage Finish'

1. **PID**: Parcel Identification Number
2. **MS SubClass**: Identifies the type of dwelling involved in the sale
3. **MS Zoning**: Identifies the general zoning classification of the sale
4. **Lot Frontage**: Linear feet of street connected to property
5. **Lot Area**: Lot size in square feet
6. **Street**: Type of road access to property
7. **Alley**: Type of alley access to property
8. **Lot Shape**: General shape of property
9. **Land Contour**: Flatness of the property
10. **Utilities**: Type of utilities available
11. **Lot Config**: Lot configuration
12. **Land Slope**: Slope of property
13. **Neighborhood**: Physical locations within Ames city limits
14. **Condition 1**: Proximity to various conditions
15. **Condition 2**: Proximity to various conditions (if more than one is present)
16. **Bldg Type**: Type of dwelling
17. **House Style**: Style of dwelling
18. **Overall Qual**: Overall material and finish quality
19. **Overall Cond**: Overall condition rating
20. **Year Built**: Original construction date
21. **Year Remod/Add**: Remodel date (same as construction date if no remodeling or additions)
22. **Roof Style**: Type of roof
23. **Roof Matl**: Roof material
24. **Exterior 1st**: Exterior covering on house
25. **Exterior 2nd**: Exterior covering on house (if more than one material)
26. **Mas Vnr Type**: Masonry veneer type
27. **Mas Vnr Area**: Masonry veneer area in square feet
28. **Exter Qual**: Exterior material quality
29. **Exter Cond**: Present condition of the material on the exterior
30. **Foundation**: Type of foundation
31. **Bsmt Qual**: Height of the basement
32. **Bsmt Cond**: General condition of the basement
33. **Bsmt Exposure**: Walkout or garden level basement walls
34. **BsmtFin Type 1**: Quality of basement finished area
35. **BsmtFin SF 1**: Type 1 finished square feet
36. **BsmtFin Type 2**: Quality of second finished area (if present)
37. **BsmtFin SF 2**: Type 2 finished square feet
38. **Bsmt Unf SF**: Unfinished square feet of basement area
39. **Total Bsmt SF**: Total square feet of basement area
40. **Heating**: Type of heating
41. **Heating QC**: Heating quality and condition
42. **Central Air**: Central air conditioning
43. **Electrical**: Electrical system
44. **1st Flr SF**: First floor square feet
45. **2nd Flr SF**: Second floor square feet
46. **Low Qual Fin SF**: Low quality finished square feet (all floors)
47. **Gr Liv Area**: Above grade (ground) living area square feet
48. **Bsmt Full Bath**: Basement full bathrooms
49. **Bsmt Half Bath**: Basement half bathrooms
50. **Full Bath**: Full bathrooms above grade
51. **Half Bath**: Half baths above grade
52. **Bedroom AbvGr**: Number of bedrooms above basement level
53. **Kitchen AbvGr**: Number of kitchens
54. **Kitchen Qual**: Kitchen quality
55. **TotRms AbvGrd**: Total rooms above grade (does not include bathrooms)
56. **Functional**: Home functionality rating
57. **Fireplaces**: Number of fireplaces
58. **Fireplace Qu**: Fireplace quality
59. **Garage Type**: Garage location
60. **Garage Yr Blt**: Year garage was built
61. **Garage Finish**: Interior finish of the garage
62. **Garage Cars**: Size of garage in car capacity
63. **Garage Area**: Size of garage in square feet
64. **Garage Qual**: Garage quality
65. **Garage Cond**: Garage condition
66. **Paved Drive**: Paved driveway
67. **Wood Deck SF**: Wood deck area in square feet
68. **Open Porch SF**: Open porch area in square feet
69. **Enclosed Porch**: Enclosed porch area in square feet
70. **3Ssn Porch**: Three season porch area in square feet
71. **Screen Porch**: Screen porch area in square feet
72. **Pool Area**: Pool area in square feet
73. **Pool QC**: Pool quality
74. **Fence**: Fence quality
75. **Misc Feature**: Miscellaneous feature not covered in other categories
76. **Misc Val**: Value of miscellaneous feature
77. **Mo Sold**: Month Sold
78. **Yr Sold**: Year Sold
79. **Sale Type**: Type of sale
80. **Sale Condition**: Condition of sale
81. **SalePrice**: Sale price

In [23]:
data['Sale Type'].value_counts()

Sale Type
WD       2536
New       239
COD        87
ConLD      26
CWD        12
ConLI       9
ConLw       8
Oth         7
Con         5
VWD         1
Name: count, dtype: int64